daffodil-runtime1/src/main/scala/org/apache/daffodil/dpath/NodeInfo.scala

```diff
@@ -594,7 +617,7 @@ object NodeInfo extends Enum {
     protected sealed trait NonNegativeIntegerKind extends Integer.Kind
     case object NonNegativeInteger extends PrimTypeNode(Integer, List(UnsignedLong)) with NonNegativeIntegerKind with PrimNumeric {
       type Kind = NonNegativeIntegerKind
-      override def fromXMLString(s: String): DataValueBigInt = new JBigInt(s)
+      override def fromString(s: String): DataValueBigInt = new JBigInt(s)
```

Comment:
Looks like we aren't actually validating that the string is negative? We should add some range checking.

In fact, looks like all of our unsinged primitives don't validate that result of fromXMLString is unsigned. Should we add an ``isValidRange`` check in ``fromXMLString``, or maybe just have each of the unsigned prims manually check inside fromString?

Reply:
This isValidRange is called elsewhere in number parse primitives. Arguably, it could be checked here and exception thrown at the point where we synthesize the type, and then all the places currently calling isValidRange would instead have to catch the exception . 

I think the design principle here is that we don't have context to report diagnostics properly here, so we do as little checking as possible here, and provide check calls that the calling context can use, assuming that it has better contextual information for diagnostics.

Reply:
I'm not saying that this is a *good* design point, I think probably it would be cleaner contract if these types succeded and produced a value of the right type in the right range, or throw an appropriate exception. But that will require quite a bit of refactoring, will change some diagnostic messages, etc.

Reply:
True, we aren't consistent about this at all right now. It's probably worth putting that change off, as you're right, it probably would be a pretty large refactor.

Reply:
I am now rolling in changes for DAFFODIL-2338 into this pull request. Ended up not needing such a large refactor.